packages/rocketchat-api/server/v1/channels.js

```diff
@@ -404,8 +404,13 @@ RocketChat.API.v1.addRoute('channels.list', { authRequired: true }, {
 RocketChat.API.v1.addRoute('channels.list.joined', { authRequired: true }, {
 	get() {
 		const { offset, count } = this.getPaginationItems();
-		const { sort, fields } = this.parseJsonQuery();
-		let rooms = _.pluck(RocketChat.models.Subscriptions.findByTypeAndUserId('c', this.userId).fetch(), '_room');
+		const { sort, fields, query } = this.parseJsonQuery();
+		const ourQuery = Object.assign({}, query, {
+			t: 'c',
+			'u._id': this.userId
+		});
+
+		let rooms = _.pluck(RocketChat.models.Subscriptions.find(ourQuery).fetch(), '_room');
```

Comment:
Is it safe to just blindly pass query through?

Reply:
@geekgonecrazy it's not exactly passing the query blindly. The function `parseJsonQuery` handles several cases and line 410 ensures that the calling user is only able to query their own...although I say that without trying it. @xbolshe what happens whenever someone passes the query with a nested property like the one the string `u._id` is doing? For example, `{ "u": { "_id": "other_users_id" } }`?

Reply:
@graywolf336   One string and nested string will have different actions as described here: https://stackoverflow.com/questions/16002659/how-to-query-nested-objects  
Becase "u" has several fields inside, only nested "_id" will find no records at all:

![default](https://user-images.githubusercontent.com/10796236/35251187-19f58046-ffeb-11e7-82bb-94971aa9a4f4.png)

But one string provides a records selection:

![default](https://user-images.githubusercontent.com/10796236/35251219-3f43f350-ffeb-11e7-9bed-074b83163f36.png)

So, need to use all fields nested inside "u" like shown below:

![default](https://user-images.githubusercontent.com/10796236/35251313-b7e0ba5a-ffeb-11e7-825d-d35db7ed0801.png)

Providing 2 different values for one field like "u._id" will cause a search of no records:

![default](https://user-images.githubusercontent.com/10796236/35251389-09c37790-ffec-11e7-962a-ce7c78d56e61.png)

Answeing on your question, one field cannot be equal at the same time to 2 different values. 

If a value will be duplicated like shown below, a search will be performed:

![default](https://user-images.githubusercontent.com/10796236/35251588-f46bf4c0-ffec-11e7-87d5-19ae3dad31f8.png)

So, only records belongs to this.userId will be searched for channels.list.joined.
I have tried to inherit constraints like this.userId applied before my changes.

Reply:
BTW: the last constraint overwrites the previous one:
![default](https://user-images.githubusercontent.com/10796236/35252151-1c3eebb2-fff0-11e7-9955-56a32698e396.png)

![default](https://user-images.githubusercontent.com/10796236/35252168-2a290294-fff0-11e7-90fc-76658628a636.png)

So, 'u._id': this.userId shall be after 'query' as it's done.

Reply:
Hello Devs, 
Still I'm not getting the expected result of this , I'm trying to pass the query parameter for this [https://ctsichat.mvoipctsi.com/api/v1/groups.list?query= { "customFields.field1": { "$regex": "Coders"}} ](url) to filter the list on Coders group but all groups are  getting in the response , so using this query I expecting below result, 

{
            "name": "Coders",
            "fname": "Coders",
            "t": "p",
            "msgs": 0,
            "u": {
                "_id": "LgHtJ3nDioaStn2dy",
                "username": "sachin"
            },
            "customFields": {
                "field1": "AndroidUsers"
            },
            "ts": "2018-01-24T08:04:45.372Z",
            "ro": false,
            "sysMes": true,
            "_updatedAt": "2018-01-24T08:04:45.372Z",
            "_id": "w5SRcgYTvwjxBaWZF",
            "usernames": [
                "sachin"
            ]
        }, 

Is their any solution, what i am doing wrong, please help me out.

Reply:
Thanks for looking into it in more depth! :+1: